src/EmbeddedManifold.jl

```diff
@@ -228,400 +228,183 @@ This is used by the [`AbstractDecoratorManifold`](@ref) within
-    return Val(:intransparent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(embed!),
-    ::AbstractEmbeddedManifold,
-    args...,
-)
-    return Val(:intransparent)
-end
-function decorator_transparent_dispatch(::typeof(exp), ::AbstractEmbeddedManifold, args...)
-    return Val(:parent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(exp),
-    ::AbstractEmbeddedManifold{𝔽,<:TransparentIsometricEmbedding},
-    args...,
-) where {𝔽}
-    return Val(:transparent)
-end
-function decorator_transparent_dispatch(::typeof(exp!), ::AbstractEmbeddedManifold, args...)
-    return Val(:intransparent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(exp!),
-    ::AbstractEmbeddedManifold{𝔽,<:TransparentIsometricEmbedding},
-    args...,
-) where {𝔽}
-    return Val(:transparent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(get_basis),
-    ::AbstractEmbeddedManifold,
-    args...,
-)
-    return Val(:parent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(get_coordinates),
-    ::AbstractEmbeddedManifold,
-    args...,
-)
-    return Val(:parent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(get_coordinates!),
-    ::AbstractEmbeddedManifold,
-    args...,
-)
-    return Val(:parent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(get_vector),
-    ::AbstractEmbeddedManifold,
-    args...,
-)
-    return Val(:parent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(get_vector!),
-    ::AbstractEmbeddedManifold,
-    args...,
-)
-    return Val(:parent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(inner),
-    ::AbstractEmbeddedManifold,
-    args...,
-)
-    return Val(:intransparent)
-end
-function decorator_transparent_dispatch(
-    ::typeof(inner),
-    ::AbstractEmbeddedManifold{𝔽,<:AbstractIsometricEmbeddingType},
-    args...,
-) where {𝔽}
-    return Val(:transparent)
+default_embedding_dispatch(::AbstractEmbeddedManifold) = Val(false)
+
+#
+# Abstract intransparent – i.e. new implementations necessary
+for f in
+    [check_point, check_vector, embed!, exp!, inner, log!, manifold_dimension, project!]
+    eval(
+        quote
+            function decorator_transparent_dispatch(
+                ::typeof($f),
+                ::AbstractEmbeddedManifold,
+                args...,
+            )
+                return Val(:intransparent)
+            end
+        end,
+    )
 end
-function decorator_transparent_dispatch(
-    ::typeof(inverse_retract),
-    ::AbstractEmbeddedManifold,
-    args...,
-)
-    return Val(:parent)
+#
+# Abstract parent – i.e. pass to embedding
+for f in [
+    embed,
+    get_basis,
+    get_coordinates,
+    get_coordinates!,
+    get_vector,
+    get_vector!,
+    inverse_retract!,
+    mid_point!,
+    project,
+    retract!,
+    vector_transport_along,
+    vector_transport_direction,
+    vector_transport_to,
```

Comment:
Why, for some functions, does this list have mutating variants but not for other?

Reply:
BTW, making this dispatch statements in a loop is a great idea :slightly_smiling_face: .

Reply:
I id not change the behaviour here, that is just what we had before.
Note that embed! is done below, as are the vector transports...for the others I am not sure, why we did it that way before.

Thanks – reduced the code by nearly 100 loc.

Reply:
I'd say that, as long as the tests pass, making this change is a good idea and we can figure out better function lists later.